docs: update ROOK_OBC_WATCH_OPERATOR_NAMESPACE in obc doc (#5679)
rook/rook: Documentation/ceph-object-bucket-claim.md

- an `Object Bucket (OB)` is a custom resource automatically generated when a bucket is provisioned. It is a global resource, typically not visible to non-admin users, and contains information specific to the bucket. It is described by an OB CRD, also shown below.

An OBC references a storage class which is created by an administrator. The storage class defines whether the bucket requested is a new bucket or an existing bucket. It also defines the bucket retention policy.

> Users request a new or existing bucket by creating an OBC which is shown below. The ceph provisioner detects the OBC and creates a new bucket or grants access to an existing bucket, depending the the storage class referenced in the OBC. It also generates a Secret which provides credentials to access the bucket, and a ConfigMap which contains the bucket's endpoint. Application pods consume the information in the Secret and ConfigMap to access the bucket. Please note to make provisioner watch only the rook operator namespace set `ROOK_OBC_WATCH_OPERATOR_NAMESPACE` in the operator manifest, otherwise it uses the namespace of the cluster.

Reviewer: ```suggestion
Users request a new or existing bucket by creating an OBC which is shown below. The Ceph provisioner detects the OBC and creates a new bucket or grants access to an existing bucket, depending on the storage class referenced in the OBC. It also generates a Secret that provides credentials to access the bucket and a ConfigMap which contains the bucket's endpoint. Application pods consume the information in the Secret and ConfigMap to access the bucket. Please note that to make the provisioner watch the cluster namespace only you need to set `ROOK_OBC_WATCH_OPERATOR_NAMESPACE` to `true` in the operator manifest, otherwise, it watches all namespaces.
```

Author: Done
